### DISCUSSION TITLE
Docker image - allow for sub packages in mounted volume `data/extensions`

### DISCUSSION THREAD

**Discussion by gnimmelf** [directus/directus#5172] [Feature Requests]
Hi,

I'ts late at nite here, and I'm suggesting this after minimal research, but;

Using the docker image, I'm triggering a Cloudflare Pages rebuild through `extensions/hooks`. The `docker-compose.yaml` and `extensions`-folder are part of a git-repo. 

Would it be possible to use s.th like lerna / [subpkg](https://www.npmjs.com/package/subpkg) to get dependencies loaded for the extensions-folder by including a `package.json` there? - And you guys including 'sub package loader' in the docker image...

Cheers,
Flemming

**Comment by gnimmelf** [comment]
Heeeh,

Just place your `node_modules` directory into the extensions-folder...?
 -If so, pointing that out would make a great addition to the documentation =D

**Reply by rijkvanzanten** [reply]
Yup! You can add whatever you like into extensions, as long as the main "entrypoint" is a regular JS file that the api/app can read 👍🏻

**Reply by andriusign** [reply]
@rijkvanzanten  Please provide a working example of Dockerfile for building custom Directus image with my own extensions and endpoints.
This does not work as expected:

```
# Stage 1
FROM directus/directus:v9.0.0-rc.51

RUN mkdir -p /directus

WORKDIR /directus/

RUN cd ./extensions

RUN npm install

COPY ./extensions /directus/extensions
```

Output:
11:21:43 ⚠️ Couldn't register endpoint "article"
11:21:43 ⚠️ Cannot find module 'moment-timezone'
Require stack:

/directus/extensions/lib/article-publish.service.js
/directus/extensions/endpoints/article/index.js
/directus/node_modules/directus/dist/extensions.js
/directus/node_modules/directus/dist/controllers/extensions.js
/directus/node_modules/directus/dist/app.js
/directus/node_modules/directus/dist/server.js
/directus/node_modules/directus/dist/start.js
/directus/node_modules/directus/dist/cli/index.js
/directus/node_modules/directus/cli.js

Not sure about paths and how you manage then, it is a bit confusing where and what has to be.

**Reply by andriusign** [reply]
File structure:
![image](https://user-images.githubusercontent.com/14263959/116781073-3d436300-aa89-11eb-81cd-6e3d452ce948.png)

**Comment by andriusign** [comment]
Dockerfile that worked for me:

```
FROM node:14-alpine as build-step

WORKDIR /directus/extensions

COPY ./extensions .

RUN npm install

# Stage 2
FROM directus/directus:v9.0.0-rc.51

WORKDIR /directus

COPY --from=build-step /directus/extensions /directus/extensions

```

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
